README.md

```diff
@@ -100,46 +96,108 @@ const useProduct = createRemoteResource({
+#### Resource
+
+The return value from `createRemoteResource` has the following shape:
+
+```ts
+{
+  id: string,
+  createEntryId: (...args: Array<any>) => string,
+  initialValue: any,
+  invalidateAfter: number,
+  load: (...args: Array<any>) => Promise<any>,
+  save: (...args: Array<any>) => Promise<any>,
+  delete: (...args: Array<any>) => Promise<any>,
+  getEntry: string => Immutable.RecordOf<{
+    id: string,
+    data: Maybe<any>,
+    updatedAt: Maybe<number>, // Unix timestamp
+    loadPromise: Maybe<Promise<any>>
+  }>,
+  onChange: (() => void) => void, // Allows for subscribing to state changes. Basically a wrapper around store.subscribe.
+  dispatch: ({ type: string }) => void // store.dispatch that adds `resourceId` to the action payload
+};
+```
+
+### `useResourceState`
+
+A React hook that takes a resource and an optional array of arguments and returns a tuple, very much like React's `useState`. The second item in the tuple works like `useState` in that it sets the in-memory state of the resource. Unlike `useState`, however, the state is not local to the component. Any other components that are using the state of that same resource get updated immediately with the new state! Under the hood `react-remote-resource` implements a redux store. Every resource get its own state and there can be multiple entries for each resource, depending on the arguments you pass into `useResourceState` and the optional `createEntryId` option that you can pass into `createRemoteResource`. By default every new combination of arguments will create a new entry in the store.
 
 ```jsx
-const [product, actions] = useProduct(productId);
+import { useResourceState } from "react-remote-resource";
+
+const ProductCategory = ({ categoryId }) => {
+  const [categoriesById] = useResourceState(categoriesState);
+  const [products] = useResourceState(productsResource, [categoryId]);
+
+  return (
+    <section>
+      <header>
+        <h1>{categoriesById[categoryId].name}</h1>
+      </header>
+      <ul>
+        {products.map(product => (
+          <li key={product.id}>
+            <Link to={`/products/${product.id}`}>{product.name}</Link>
+          </li>
+        ))}
+      </ul>
+    </section>
+  );
+};
+```
+
+This hook is very powerful. Let's walk through what happens when it is used:
 
-// Bypasses the cache and calls `load` again but without throwing the promise. Returns the promise from `load`.
-actions.refresh();
+1. If there is no data in the store for this resource OR if there is no entry for the combination of arguments OR if the component is mounting and the last time the data was loaded for this entry was longer ago than the `invalidateAfter` argument then the `load` function that was passed into `createRemoteResource` will be invoked and the returned promise will be thrown until the promise resolves or rejects.
```

Comment:
```suggestion
1. If there is no data for this resource OR no entry in the cache for the arguments OR the cache is stale (See [invalidateAfter](https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource)), then the `load` function of the `categoriesResource` and `productsResource` will be invoked and promises thrown.
```

Reply:
Nice! I added "(See [invalidateAfter](https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource))" after "no entry in the cache for the arguments" because I thought it was relevant.